src/avt/FileWriter/avtDatasetFileWriter.C

```diff
@@ -830,10 +837,14 @@ avtDatasetFileWriter::WriteCurveFile(const char *filename, int quality, int comp
     vtkPoints *pts = pd->GetPoints();
     for (size_t i = 0 ; i < line_segments.size() ; i++)
     {
-        if (line_segments.size() <= 1)
-            ofile() << varTag << " curve" << endl;
-        else
+        // Prefer labels if present
+        if (!labels.empty())
+            ofile() << varTag << " " << labels[i] << endl;
+        // Default to generic 'curve'
+        else if (line_segments.size() > 1)
             ofile() << varTag << " curve" << i << endl;
+        else
+            ofile() << varTag << " curve" << endl;
 
```

Comment:
How big is the labels array likely to be?

Reply:
Mostly just 1. Pick could have multiple, and it would depend on the number of variables.

Reply:
How big is `line_segments`?

Reply:
# of curves, generally 1, multiples if time-pick requested multiple variables.

Reply:
just to clarify, line_segments.size() should equal labels.size() and indicates number of curves being drawn.

Reply:
So the loop isn't happening a bunch of times. If it was happening a lot I might suggest pulling `if (!labels.empty())` out above the loop and making two loops.

Reply:
Would it make sense to use `bool useLabels = !labels.empty();` outside the for loop. Then use the bool for the test inside the for loop. Could do the same for the `line_segments.size() > 1` test.

Reply:
Yes that is a good idea.